hack/generated/controllers/generic_controller.go

```diff
@@ -312,16 +308,19 @@ func (gr *GenericReconciler) StartDeleteOfResource(
 		return ctrl.Result{}, errors.Wrapf(err, "couldn't convert to armResourceSpec")
 	}
 
+	var retryAfter time.Duration // ARM can tell us how long to wait for a DELETE
+
 	err = gr.Patch(ctx, data, func(ctx context.Context, mutData *ReconcileMetadata) error {
 		emptyStatus, err := reflecthelpers.NewEmptyArmResourceStatus(mutData.metaObj)
 		if err != nil {
 			return errors.Wrapf(err, "creating empty status for %q", resource.GetId())
 		}
 
-		err = gr.ARMClient.BeginDeleteResource(ctx, resource.GetId(), resource.Spec().GetApiVersion(), emptyStatus)
+		err, retryAfter = gr.ARMClient.BeginDeleteResource(ctx, resource.GetId(), resource.Spec().GetApiVersion(), emptyStatus)
```

Comment:
Do we care to be able to distinguish "there was no retryAfter header" from "there was a retryAfter header and it was zero"?

I'm thinking it'd be sorta easy to call this and think that the service was saying to retry immediately when really it just didn't give any guidance.

Reply:
Or maybe we're ok with saying that even if the service gave `0` as their retryAfter we just wouldn't honor it.

Reply:
@matthchr that was my thought 🙂

Reply:
Fine by me